### PR TITLE
chore: delete storage transfer samples

### LINKED ISSUE #2148: storage-transfer: "before all" hook for "should create a storage transfer job" failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: eabc6eacce2441ae9096dfca7d46816370c7a840
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/52b6d1d1-b530-49cd-819f-481037eaf4a0), [Sponge](http://sponge2/52b6d1d1-b530-49cd-819f-481037eaf4a0)
status: failed
<details><summary>Test output</summary><br><pre>Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)</pre></details>

### PR DESCRIPTION
**Pull request by tritone**
Reasons to delete:
* Added in 2016 and not substantially updated since.
* Region tags were never implemented in other languages or used in the docs.
* Does not conform to current sample standards.
* @JesseLovelace started a new set of updated, conforming samples for STS (Java and Python implemented/published so far).

Fixes #2148

### CODE REVIEW

**Comment by tritone** [comment]
@fhinkel is there something I missed as far as removing this from the CI?

**Comment by fhinkel** [comment]
I removed it from `required` tests. Did you send an internal CL to remove the job configs?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ need to enable the appropriate APIs in the [Cloud Console](https://console.cloud
 
         gcloud auth application-default login
 
-1. Change directory to one of the sample folders, e.g. `storage-transfer`.
+1. Change directory to one of the sample folders, e.g. `datastore`.
 
-        cd storage-transfer/
+        cd datastore/
```
Comment (fhinkel) [inline CONTRIBUTING.md:14]: Good catch! Thank you.

**Review by fhinkel — APPROVED** [review]
Great, thanks!

**Pull request merged**